server/neptune/gateway/event/preworkflow/runner.go

```diff
@@ -0,0 +1,39 @@
+package preworkflow
+
+import (
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/core/config/valid"
+	"github.com/runatlantis/atlantis/server/events/models"
+)
+
+type executor interface {
+	Execute(hook *valid.PreWorkflowHook, repo models.Repo, path string) error
+}
+
+type HooksRunner struct {
+	GlobalCfg    valid.GlobalCfg
+	HookExecutor executor
+}
+
+func (r *HooksRunner) Run(baseRepo models.Repo, repoDir string) error {
```

Comment:
Rather than clone within the runner, I'd rather let the caller handle the cloning and pass its location to the hooks runner.